imputation_functions.py

```diff
@@ -51,61 +79,58 @@ def factors_calculation_a(row, questions, parameters):
 
     :return: row of DataFrame
     """
-    # extract the parameters:
-    first_threshold = parameters["first_threshold"]
-    second_threshold = parameters["second_threshold"]
-    third_threshold = parameters["third_threshold"]
-    first_imputation_factor = parameters["first_imputation_factor"]
-    second_imputation_factor = parameters["second_imputation_factor"]
-    third_imputation_factors = parameters["third_imputation_factors"]
-    region_column = parameters["region_column"]
-    regionless_code = parameters["regionless_code"]
-    survey_column = parameters["survey_column"]
-    percentage_movement = parameters["percentage_movement"]
-    distinct_values = parameters["distinct_values"]
+
+    runtime_variables, errors = ExtendedFactorsCalculationASchema().load(kwargs)
+    if errors:
+        raise ValueError(f"Error validating factors params: {errors}")
+
+    rv = SimpleNamespace(**runtime_variables)
```

Comment:
You are going to hate me, but this is the only file where you abbreviated runtime_variables to RV and it stands out.

Reply:
Urgh you're right. I'll change it to runtime_object as that what it is.